### PR TITLE
chore: tsconfig cleanup

### PR DESCRIPTION
**Pull request by kanadgupta**
## 🧰 Changes

Cleans up and modernizes our `tsconfig`!

## 🧬 QA & Testing

If the build passes across the board we should be good to go.

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -2,20 +2,14 @@
   "compilerOptions": {
     "allowJs": true,
     "baseUrl": "./src",
-    "downlevelIteration": true,
-    "esModuleInterop": true,
```
Comment (kanadgupta) [inline tsconfig.json:6]: this flag is enabled by default now that we're on `NodeNext`

File: tsconfig.json [hunk 2]
```diff
@@ -2,20 +2,14 @@
...
     "experimentalDecorators": true,
-    "forceConsistentCasingInFileNames": true,
```
Comment (kanadgupta) [inline tsconfig.json:8]: this flag is enabled by default

File: tsconfig.json [hunk 3]
```diff
@@ -2,20 +2,14 @@
...
-    "lib": ["ES2020", "ES2021.Intl"],
+    "lib": ["ES2022"],
     "module": "NodeNext",
     "outDir": "dist/",
-    "paths": {
-      "editor": [".sink.d.ts"]
-    },
```
Comment (kanadgupta) [inline tsconfig.json:14]: whoops

File: tsconfig.json [hunk 4]
```diff
@@ -2,20 +2,14 @@
...
     "resolveJsonModule": true,
```
Comment (erunion) [inline tsconfig.json:9]: Are we importing JSON files anywhere not in a unit test?

**Review by erunion — APPROVED** [review]

File: tsconfig.json (hunk 4 above)
Comment (kanadgupta) [reply tsconfig.json:9]: yeah we use our `package.json` file in a bunch of places

**Pull request merged**